flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/ChangelogPlanUtils.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.utils
+
+import org.apache.flink.table.connector.ChangelogMode
+import org.apache.flink.table.planner.plan.`trait`.{ModifyKind, ModifyKindSetTraitDef, UpdateKind, UpdateKindTraitDef}
+import org.apache.flink.table.planner.plan.nodes.physical.stream.StreamPhysicalRel
+import org.apache.flink.table.planner.plan.optimize.program.FlinkChangelogModeInferenceProgram
+import org.apache.flink.types.RowKind
+import org.apache.calcite.rel.RelNode
+
+import scala.collection.mutable.ArrayBuffer
+
+/**
+ * Utilities for changelog plan.
+ */
+object ChangelogPlanUtils {
+
+  /**
+   * Returns true if the plan produces insert-only changes.
+   *
+   *  <p>Note: this method must be called after [[FlinkChangelogModeInferenceProgram]] is applied.
+   */
+  def isInsertOnly(plan: RelNode): Boolean = {
+    assert(plan.isInstanceOf[StreamPhysicalRel])
```

Comment:
I still would like to keep this parameter as `RelNode`, otherwise, we will have a lot of duplicate codes, e.g. `getInput().asInstanceOf[StreamPhysicalNode]`. As this is an internal method, I think it's find to have an assert here to reduce the duplicate codes. 

For other methods in this class, I changed to `StreamPhysicalRel` parameter.